src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,634 @@
+  // !-----------------------------------------------------------------------
+
+  // @param [in]  imode mode index
+  // @param [in]  isprx_kk if there is precip
+  // @param [in]  dgn_awet_imode_kk ! wet aerosol diameter of mode imode at
+  // elevation kk [m]
+  // @param [out] scavcoefnum scavenging removal for aerosol number [1/h]
+  // @param [out] scavcoefvol scavenging removal for aerosol volume [1/h]
+
+  // NOTE: original FORTRAN function has two internal loops: kk and icol.
+  // We removed these loops. Hence, the inputs/outputs of
+  // modal_aero_bcscavcoef_get are reals at kk , icol locations
+
+  const Real zero = 0;
+  const Real one = 1;
+  // BAD CONSTANT
+  const Real dlndg_nimptblgrow = haero::log(1.25);
+  if (isprx_kk) {
+    // ! interpolate table values using log of
+    // (actual-wet-size)/(base-dry-size) ratio of wet and dry aerosol diameter
+    // [fraction]
+    const Real wetdiaratio = dgn_awet_imode_kk / dgnum_amode_imode;
+    // BAD CONSTANT
+    Real scavimpvol, scavimpnum = zero;
+    if (wetdiaratio >= 0.99 && wetdiaratio <= 1.01) {
+      scavimpvol = scavimptblvol[-nimptblgrow_mind][imode];
+      scavimpnum = scavimptblnum[-nimptblgrow_mind][imode];
+    } else {
+      Real xgrow = haero::log(wetdiaratio) / dlndg_nimptblgrow;
+      int jgrow = int(xgrow);
+      if (xgrow < zero) {
+        jgrow = jgrow - 1;
+      }
+
+      if (jgrow < nimptblgrow_mind) {
+        jgrow = nimptblgrow_mind;
+        xgrow = jgrow;
+      } else {
+        jgrow = haero::min(jgrow, nimptblgrow_maxd - 1);
+      }
+      const Real dumfhi = xgrow - jgrow;
+      const Real dumflo = one - dumfhi;
+      // Fortran to C++ index conversion
+      // note that nimptblgrow_mind is negative (-7)
+      int jgrow_pp = jgrow - nimptblgrow_mind;
+      scavimpvol = dumflo * scavimptblvol[jgrow_pp][imode] +
+                   dumfhi * scavimptblvol[jgrow_pp + 1][imode];
+      scavimpnum = dumflo * scavimptblnum[jgrow_pp][imode] +
+                   dumfhi * scavimptblnum[jgrow_pp + 1][imode];
+
+    } /// wetdiaratio
+      // ! impaction scavenging removal amount for volume
+    scavcoefvol_kk = haero::exp(scavimpvol);
+    // ! impaction scavenging removal amount to number
+    scavcoefnum_kk = haero::exp(scavimpnum);
+  } else {
+    scavcoefvol_kk = zero;
+    scavcoefnum_kk = zero;
+  } // isprx_kk
+
+} // modal_aero_bcscavcoef_get
+
+KOKKOS_INLINE_FUNCTION
+Real air_dynamic_viscosity(const Real temp) {
+  /*-----------------------------------------------------------------
+  ! Calculate dynamic viscosity of air, unit [g/cm/s]
+  !
+  ! note that this calculation is different with that used in dry deposition
+  ! see the same-name function in modal_aero_drydep.F90
+  !-----------------------------------------------------------------*/
+  // @param [in] temp   ! air temperature [K]
+  // @return     dynamic viscosity of air, unit [g/cm/s]
+  return 1.8325e-4 * (416.16 / (temp + 120.)) * haero::pow(temp / 296.16, 1.5);
```

Comment:
Can we provide a citation for this expression?

Reply:
I add this web page (pdf) in teh code as reference .